Source/WebGPU/WebGPU/UsdModelRenderer.swift

```diff
@@ -0,0 +1,428 @@
+        let captureDescriptor = MTLCaptureDescriptor()
+        captureDescriptor.captureObject = device
+        captureDescriptor.destination = .gpuTraceDocument
+        let outputURL = URL.temporaryDirectory.appending(path: "capture_\(dateString).gputrace").standardizedFileURL
+        captureDescriptor.outputURL = outputURL
+        return (captureDescriptor, outputURL)
+    }
+}
+
+@objc
+@implementation
+extension DDBridgeReceiver {
+    fileprivate var device: MTLDevice?
+
+    #if canImport(DirectDrawBackend)
+    @nonobjc
+    fileprivate var context: DirectDrawBackend.Context?
+    @nonobjc
+    fileprivate var scene: DirectDrawBackend.Scene?
+    @nonobjc
+    fileprivate var meshes: [UUID: DirectDrawBackend.MeshInstance] = [:]
+    @nonobjc
+    fileprivate var materials: [UUID: DirectDrawBackend.MaterialInstance] = [:]
+    // var textures: [UUID:(DirectDrawBackend.TextureResource, String)] = [:]
+    // var textureFromPath: [String:UUID] = [:]
+    // @nonobjc fileprivate var nodeDefinitionStore: ShaderGraph.NodeDefinitionStore?
+    @nonobjc
+    fileprivate var camera: Camera?
+    #endif
+    @nonobjc
+    fileprivate var commandQueue: MTLCommandQueue?
+    @nonobjc
+    fileprivate var captureManager: MTLCaptureManager?
+
+    @objc
+    override init() {
+        super.init()
+    }
+    @objc(setDeviceWithDevice:)
+    func setDeviceWith(_ device: MTLDevice) {
+        self.device = device
+
+        #if canImport(DirectDrawBackend)
+        self.context = DirectDrawBackend.Context.createContext(device: device)
+        self.scene = self.context?.makeScene()
+        // self.nodeDefinitionStore = NodeDefinitionStore()
+        self.commandQueue = device.makeCommandQueue()
+        self.captureManager = MTLCaptureManager.shared()
+        self.camera = context?.makeCamera()
+        self.context?.setCameraDistance(171.0)
+        self.context?.setEnableModelRotation(true)
+        #endif
+    }
+
+    @objc(renderWithTexture:)
+    func render(with texture: MTLTexture) {
+        let (captureDescriptor, outputURL) = Helper.makeCaptureDescriptor(device: self.device)
+        do {
+            try captureManager?.startCapture(with: captureDescriptor)
+        } catch {
+            print("failed to start gpu capture")
+        }
+
+        #if canImport(DirectDrawBackend)
+
+        self.context?.update(deltaTime: 1.0 / 60.0)
+        self.camera?.renderTarget = .texture(texture)
+        guard let scene = self.scene else {
+            return
+        }
+        guard let camera = self.camera else {
+            return
+        }
+        self.context?.render(scene: scene, camera: camera)
+
+        print("Captured gpu trace at \(outputURL)")
+        captureManager?.stopCapture()
+        #endif
+    }
+
+    @objc(addMesh:identifier:)
+    func addMesh(_ descriptor: DDBridgeAddMeshRequest, identifier: UUID) -> Bool {
+        #if canImport(DirectDrawBackend)
+        let meshDescriptor = descriptor.makeDescriptor()
```

Comment:
descriptor isnt really a descriptor isnt it just a request which has a descriptor on it?

Reply:
`descriptor` is our descriptor and `meshDescriptor` is RealityKit's